src/client/features/view/components/tooltips/generateContent.js

```diff
@@ -0,0 +1,238 @@
+
+
+//Handle name related metadata fields
+const standardNameHandler = (pair) => makeTooltipItem(pair[1], 'Approved Name: ');
+const displayNameHandler = (pair) => makeTooltipItem(pair[1], 'Display Name: ');
+const nameHandler = (pair, trim) => {
+  //Trim results to first 3 names to avoid overflow
+  let shortArray = pair[1];
+  if (trim) { shortArray = pair[1].slice(0, 3); }
+
+  //Filter out Chemical formulas
+  if (shortArray instanceof Array) shortArray = shortArray.filter(name => (!name.trim().match(/^([^J][0-9BCOHNSOPrIFla@+\-\[\]\(\)\\=#$]{6,})$/ig)));
+
+  //Determine render value
+  let renderValue = h('div.tooltip-value', shortArray.toString());
+  if (!(trim)) {
+    renderValue = makeList(shortArray);
+  }
+
+  return h('div.fake-paragraph', [h('div.field-name', 'Synonyms: '), renderValue]);
+};
+
+//Handle database related fields
+const dataSourceHandler = (pair) => {
+  let source = pair[1].replace('http://pathwaycommons.org/pc2/', '');
+  let link = generateDataSourceLink(source, 'Data Source: ');
+  return h('div.fake-paragraph', link);
+};
+const databaseHandler = (pair, trim) => {
+  //Sort the array by database names
+  let sortedArray = sortByDatabaseId(pair[1]);
+  if (sortedArray.length < 1) { return; }
+  return h('div.fake-paragraph',
+    [
+      h('div.field-name', 'Database References:'),
+      h('div.wrap-text', h('ul.db-list', sortedArray.map(item => generateIdList(item, trim), this)))
+    ]);
+};
+
+//Handle type related fields
+const typeHandler = (pair, trim, key) => {
+  if (!(trim)) {
+    let type = pair[1].toString().substring(3);
+    let formattedType = type.replace(/([A-Z])/g, ' $1').trim();
+    return h('div.fake-paragraph', [h('div.field-name', key + ': '), formattedType]);
+  }
+};
+
+//Default to generating a list of all items
+const defaultHandler = (pair, trim, key) => {
+  if (trim) { return; }
+  if (key === 'Comment') { key = 'Comments'; }
+  return h('div.fake-paragraph', [
+    h('div.field-name', key + ': '),
+    makeList(pair[1])
+  ]);
+};
+
+const metaDataKeyMap = new Map()
+  .set('Standard Name', standardNameHandler)
+  .set('Display Name', displayNameHandler)
+  .set('Data Source', dataSourceHandler)
+  .set('Type', typeHandler)
+  .set('Names', nameHandler)
+  .set('Database IDs', databaseHandler);
+
+
+//Generate HTML elements for a Parsed Metadata Field
+//Optional trim parameter indicates if the data presented should be trimmed to a reasonable length
+//Data Pair -> HTML
+function parseMetadata(pair, trim = true) {
+  let key = pair[0];
+  let handler = metaDataKeyMap.get(key);
+  if (handler) {
+    return handler(pair, trim, key);
+  }
+  else {
+    return defaultHandler(pair, trim, key);
+  }
+}
+
+//Convert a generic array or string to an html list
+//Array -> HTML 
+function makeList(items, ulClass = 'ul.value-list', liClass = 'li.value-list-item') {
+  //Delete duplicates
+  items = deleteDuplicatesWithoutCase(items);
+
+  //Resolve possible errors
```

Comment:
I think all of this logic should be delegated to whoever calls makeList.

I.e create a strict contract for makeList.  Either make items be a a string or array.  Don't try to accommodate all of these different scenarios.

Reply:
Removed logic from delete duplicates and created a toHTML function which maps any given value to the correct toHTML function.